internal/controller/tokenprovider/azure_token_provider.go

```diff
@@ -18,14 +18,20 @@ import (
 
 // azureTokenProvider is a provider implements TokenProvider interface for Azure access tokens.
 type azureTokenProvider struct {
-	credential  *azidentity.ClientSecretCredential
+	credential  *azidentity.ClientAssertionCredential
 	tokenOption policy.TokenRequestOptions
 }
 
-// NewAzureTokenProvider creates a new TokenProvider with the given tenant ID, client ID, client secret, and token request options.
-func NewAzureTokenProvider(tenantID, clientID, clientSecret string, tokenOption policy.TokenRequestOptions) (TokenProvider, error) {
-	clientOptions := GetClientSecretCredentialOptions()
-	credential, err := azidentity.NewClientSecretCredential(tenantID, clientID, clientSecret, clientOptions)
+// NewAzureTokenProvider creates a new TokenProvider with the given tenant ID, client ID, tokenProvider, and token request options.
+func NewAzureTokenProvider(ctx context.Context, tenantID, clientID string, tokenProvider TokenProvider, tokenOption policy.TokenRequestOptions) (TokenProvider, error) {
+	clientOptions := GetClientAssertionCredentialOptions()
+	credential, err := azidentity.NewClientAssertionCredential(tenantID, clientID, func(_ context.Context) (string, error) {
```

Comment:
What's the difference between `NewClientSecretCredential` and `NewClientAssertionCredential`, is here to use the oidc token to exchange the azure credential ?